src/server/unit_sao.cpp

```diff
@@ -273,15 +273,26 @@ std::string UnitSAO::generateUpdateAttachmentCommand() const
 }
 
 std::string UnitSAO::generateUpdateBonePositionCommand(
-		const std::string &bone, const v3f &position, const v3f &rotation)
+		const std::string &bone, const BoneOverride *props)
 {
 	std::ostringstream os(std::ios::binary);
 	// command
 	writeU8(os, AO_CMD_SET_BONE_POSITION);
 	// parameters
 	os << serializeString16(bone);
-	writeV3F32(os, position);
-	writeV3F32(os, rotation);
+	writeV3F32(os, props->position.vector);
+	v3f euler_rot;
+	props->rotation.next.toEuler(euler_rot);
+	writeV3F32(os, euler_rot * core::RADTODEG);
+	writeV3F32(os, props->scale.vector);
+	writeF32(os, props->position.interpolation_duration);
+	writeF32(os, props->rotation.interpolation_duration);
+	writeF32(os, props->scale.interpolation_duration);
+	// clang-format off
+	writeU8(os, (props->position.absolute & 1) << 0
+	          | (props->rotation.absolute & 1) << 1
+	          | (props->scale.absolute & 1) << 2);
+	// clang-format on
```

Comment:
```suggestion
	writeU8(os, (props->position.absolute & 1) << 0
	          | (props->rotation.absolute & 1) << 1
	          | (props->scale.absolute & 1) << 2);
```

Reply:
what do you mean by that? should I remove the comments?

Reply:
yes